### PR TITLE
Add is disposed checking to single

### PR DESCRIPTION
**Pull request by s1ee**
This can cause `Undeliverable Exception` if not checked.

### CODE REVIEW

**Comment by martinbonnin** [comment]
Thanks for the contribution 👍 ! That is consistent with `Completable`, I'm not sure how come this wasn't there in the first place. 
Do you have a test case at hand that you could add to `Rx2ApolloTest`?

**Comment by martinbonnin** [comment]
@s1ee I added a unit test and the same logic for RxJava3. Sorry for the large diff, the integration tests were a bit behind and I took this opportunity to convert some of them to Kotlin.
If everything looks good for you, I'll merge this in a couple of days.

File: apollo-integration/src/test/java/com/apollographql/apollo/Rx2ApolloTest.kt [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    testObserver
+        .assertValueCount(1)
+        .assertValueAt(0) { data ->
+          assertThat(data?.hero()?.name()).isEqualTo("R2-D2")
+          true
+        }
+  }
+
+  @Test
+  fun disposingAnOperationDoesNotThrowUndeliverableException() {
```
Comment (martinbonnin) [inline apollo-integration/src/test/java/com/apollographql/apollo/Rx2ApolloTest.kt:201]: This is the test that failed before the patch ☝️

**Comment by s1ee** [comment]
Thanks for adding the tests! I was going to suggest converting the files from java to kotlin as well 🙌

File: apollo-integration/src/test/java/com/apollographql/apollo/Rx2ApolloTest.kt [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    }
+
+    val operation = apolloClient.apolloStore.write(EpisodeHeroNameQuery(Input.fromNullable(EMPIRE)), EpisodeHeroNameQuery.Data(EpisodeHeroNameQuery.Hero("", "")))
+    val testObserver = Rx2Apollo.from(operation)
+        .test()
+
+    testObserver.dispose()
+
+    // Since there is no cancellation mechanism for the ApolloStoreOperation, the only way to see if an error is thrown is to wait here
+    Thread.sleep(200)
```
Comment (s1ee) [inline apollo-integration/src/test/java/com/apollographql/apollo/Rx2ApolloTest.kt:247]: I've also used this approach to verify no error has been thrown.
https://bryanherbst.com/2020/07/15/rxjava-uncaught-exception-tests/

File: apollo-rx3-support/src/main/java/com/apollographql/apollo/rx3/Rx3Apollo.java [hunk 3]
```diff
@@ -212,12 +212,16 @@ public void subscribe(final SingleEmitter<T> emitter) {
         operation.enqueue(new ApolloStoreOperation.Callback<T>() {
           @Override
           public void onSuccess(T result) {
-            emitter.onSuccess(result);
+            if (!emitter.isDisposed()) {
```
Comment (s1ee) [inline apollo-rx3-support/src/main/java/com/apollographql/apollo/rx3/Rx3Apollo.java:215]: I totally forgot this library supports Rx3 as well 🤦

**Pull request merged**